v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstobigquery/BigtableChangeStreamsToBQIT.java

```diff
@@ -0,0 +1,520 @@
+              Assert.assertEquals(
+                  bigtableResourceManager.getInstanceId(),
+                  fvl.get(ChangelogColumn.SOURCE_INSTANCE.getBqColumnName()).getStringValue());
+              Assert.assertTrue(
+                  fvl.get(ChangelogColumn.TIEBREAKER.getBqColumnName()).getLongValue() >= 0);
+
+              toBeReadValues.remove(
+                  fvl.get(ChangelogColumn.VALUE_STRING.getBqColumnName()).getStringValue());
+            });
+
+    Storage storage = StorageOptions.newBuilder().build().getService();
+
+    long started = System.currentTimeMillis();
+    long waitForFile = Duration.ofMinutes(30).toMillis();
+
+    String filterPrefix = String.join("/", getClass().getSimpleName(), artifactClient.runId(),
+        "dlq", "severe");
+    BlobListOption listOptions = BlobListOption.prefix(filterPrefix);
+
+    boolean found = false;
+    while (!found && System.currentTimeMillis() <= (started + waitForFile)) {
+      Page<Blob> blobs = storage.list(artifactBucketName, listOptions);
+
+      for (Blob blob : blobs.iterateAll()) {
+        byte[] content = storage.readAllBytes(blob.getBlobId());
+        ObjectMapper om = new ObjectMapper();
+        JsonNode severeError = om.readTree(content);
+        Assert.assertNotNull(severeError);
+
+        String errorMessage = severeError.get("error_message").asText();
+        Assert.assertEquals(
+            "GenericData{classInfo=[errors, index], {errors=[GenericData{classInfo="
+                + "[debugInfo, location, message, reason], {reason=row-too-large}}]}}",
+            errorMessage);
+        found = true;
+      }
+
+      if (!found) {
+        Thread.sleep(1000);
+      }
+    }
+
+    if (!found) {
+      // Might be due to might be due to b/272500246
+      Assert.fail("The failed message was not found in DLQ");
+    }
+  }
+
+  @Before
+  public void setup() throws IOException {
+    bigQueryResourceManager =
+        DefaultBigQueryResourceManager.builder(testName.getMethodName(), PROJECT).build();
+    // TODO: StaticBigtableResourceManager has to be replaced with DefaultBigtableResourceManager
+    // when it supports CDC configs
+    bigtableResourceManager = StaticBigtableResourceManager.builder(/* testName.getMethodName(), */
+            PROJECT)
+        .setCredentialsProvider(credentialsProvider)
+        .setInstanceId(System.getProperty("bigtableInstanceId"))
+        .setTableId(System.getProperty("bigtableTableId"))
+        .setAppProfileId(System.getProperty("bigtableAppProfileId"))
+        .build();
+  }
+
+  @After
+  public void tearDownClass() {
+    // If we destroy Cloud Bigtable / BigQuery resources before cancelling job, it might get
+    // stuck in draining
+    if (jobInfo != null) {
+      try {
+        getDataflowClient().cancelJob(PROJECT, TEST_REGION, jobInfo.jobId());
+      } catch (Exception e) {
+        e.printStackTrace(System.err);
+      }
+    }
```

Comment:
nit: I don't think this is necessary due to the `CancelJobShutdownHook` in `launchTemplate(options)`